File: server-nestjs-hybrid/src/policies/CommentPolicy.ts
import { ResourcePolicy } from '@rhino-dev/rhino-nestjs';

/** Comment policy — HYBRID (role-free attrs; gating via group membership perms). */
export class CommentPolicy extends ResourcePolicy {
  override resourceSlug = 'comments';

  override permittedAttributesForShow(): string[] {
    return ['*'];
  }
  override hiddenAttributesForShow(): string[] {
    return [];
  }
  override permittedAttributesForCreate(): string[] {
    return ['body', 'taskId'];
  }
  override permittedAttributesForUpdate(): string[] {
    return ['body'];
  }
}

File: server-rails-multi/app/policies/comment_policy.rb
# frozen_string_literal: true

class CommentPolicy < Rhino::ResourcePolicy
  def permitted_attributes_for_show(user)
    if has_role?(user, "owner") || has_role?(user, "admin") ||
       has_role?(user, "manager") || has_role?(user, "member") ||
       has_role?(user, "viewer")
      ["*"]
    else
      []
    end
  end

  def hidden_attributes_for_show(user)
    []
  end

  def permitted_attributes_for_create(user)
    if has_role?(user, "owner") || has_role?(user, "admin") ||
       has_role?(user, "manager") || has_role?(user, "member")
      %w[body task_id]
    else
      []
    end
  end

  def permitted_attributes_for_update(user)
    if has_role?(user, "owner") || has_role?(user, "admin") ||
       has_role?(user, "manager") || has_role?(user, "member")
      %w[body]
    else
      []
    end
  end
end
